kstat.c

```diff
@@ -159,23 +159,27 @@ kstatjs_parsearg_int(const nvlist_t *arg, const char *str, int *p)
 	return (0);
 }
 
+/*
+ * This function consumes its "args" nvlist.
+ */
 static int
 kstatjs_parsearg(const nvlist_t *args, char **modulep,
     char **classp, char **namep, int *instancep)
 {
+	int rc = 0;
 	*modulep = *classp = *namep = NULL;
 	*instancep = -1;
 
 	if (kstatjs_parsearg_string(args, "module", modulep) != 0 ||
 	    kstatjs_parsearg_string(args, "class", classp) != 0 ||
-	    kstatjs_parsearg_string(args, "name", namep) != 0) {
-		return (-1);
+	    kstatjs_parsearg_string(args, "name", namep) != 0 ||
+	    kstatjs_parsearg_int(args, "instance", instancep) != 0) {
+		rc = -1;
 	}
 
-	if (kstatjs_parsearg_int(args, "instance", instancep) != 0)
-		return (-1);
-
-	return (0);
+	/* No caller uses "args" after it's done here. */
+	nvlist_free(args);
```

Comment:
This introduces a new warning:

```
../kstat.c: In function 'kstatjs_parsearg':
../kstat.c:181:21: warning: passing argument 1 of 'nvlist_free' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  181 |         nvlist_free(args);
      |                     ^~~~
In file included from /usr/include/libnvpair.h:29,
                 from ../addon.h:13,
                 from ../kstat.c:14:
/usr/include/sys/nvpair.h:156:18: note: expected 'nvlist_t *' {aka 'struct nvlist *'} but argument is of type 'const nvlist_t *' {aka 'const struct nvlist *'}
  156 | void nvlist_free(nvlist_t *);
      |                  ^~~~~~~~~~
```

Reply:
Thank you... fixing by losing "const" from `kstatjs_parsearg()` (but keeping the `const` in the ones it calls).